File: Week-4/HW-1/Steps/SearchSteps.cs
using System;
using TechTalk.SpecFlow;
using OpenQA.Selenium;

[Binding]
public class SearchSteps
{
    private readonly ScenarioContext context;

    public SearchSteps(ScenarioContext scenarioContext)
    {
        context = scenarioContext;
    }
    HomePage homePage = new HomePage();
    MenuPage menuPage = new MenuPage();
    ProductPage productPage = new ProductPage();
    SearchPage searchPage = new SearchPage();


    [Given(@"I search for (.*)")]
    public void iSearchFor(string searchKeyword)
    {
        homePage.searchBox(searchKeyword);
    }

    [When(@"I click the search button")]
    public void iClickTheSearchButton()
    {
        homePage.clickSearchButton();
    }

    [Then(@"search result should be seen '(.*)'")]
    public void SearchResultShouldBeSeen(string notification)
    {
        searchPage.verifyResult(notification);
    }


}

File: Week-5/HW-2/src/test/java/Pinterest/definations/SearchSteps.java
package Pinterest.definations;

import Pinterest.pages.HomePage;
import Pinterest.pages.SearchPage;
import io.cucumber.java.en.And;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

import java.net.MalformedURLException;


public class SearchSteps {

    HomePage homePage = new HomePage();
    SearchPage searchPage = new SearchPage();

    public SearchSteps() throws MalformedURLException {
    }

    @And("I click the search tab")
    public void iClickTheSearchTab() {
        homePage.clickSearchTab();
    }

    @When("I search for {string}")
    public void iSearchFor(String keyword) {
        searchPage.writeSearchKeyword(keyword);
    }

    @Then("I should see result {string}")
    public void iShouldSeeResult(String result) {
        searchPage.verifyResult(result);
    }
}



